### PR TITLE
move GSO control message handling to the oobConn

### PR DESCRIPTION
**Pull request by marten-seemann**
For #3999. For ECN, we'll need to append even more control messages, and the current construction would get too complicated.

### CODE REVIEW

File: sys_conn_oob.go [hunk 1]
```diff
@@ -229,7 +229,14 @@ func (c *oobConn) ReadPacket() (receivedPacket, error) {
 
 // WritePacket writes a new packet.
 // If the connection supports GSO, it's the caller's responsibility to append the right control mesage.
-func (c *oobConn) WritePacket(b []byte, addr net.Addr, oob []byte) (int, error) {
+func (c *oobConn) WritePacket(b []byte, addr net.Addr, packetInfoOOB []byte, gsoSize uint16) (int, error) {
+	oob := packetInfoOOB
```
Comment (sukunrt) [inline sys_conn_oob.go:233]: Do you need this declaration? Is is possible to just use packetInfoOOB since it is local anyway?

**Review by sukunrt — APPROVED** [review]
LGTM. Just this one comment: https://github.com/quic-go/quic-go/pull/4056#discussion_r1311180689

File: sys_conn_oob.go (hunk 1 above)
Comment (marten-seemann) [reply sys_conn_oob.go:233]: I don't, but maybe it's cleaner? I want to make it clear that the caller should only pass in the packet info control message.

File: sys_conn_oob.go [hunk 2]
```diff
@@ -229,7 +229,14 @@ func (c *oobConn) ReadPacket() (receivedPacket, error) {
 
 // WritePacket writes a new packet.
 // If the connection supports GSO, it's the caller's responsibility to append the right control mesage.
```
[the remark this answers is not in the document]
Comment (marten-seemann) [reply sys_conn_oob.go:231]: Done.

File: sys_conn_oob.go (hunk 1 above)
Comment (sukunrt) [reply sys_conn_oob.go:232]: sure. That's fine by me.

**Pull request merged**